sklearn/pipeline.py

```diff
@@ -37,6 +38,18 @@
 __all__ = ["Pipeline", "FeatureUnion", "make_pipeline", "make_union"]
 
 
+def _check_is_fitted(pipeline):
+    try:
+        check_is_fitted(pipeline)
```

Comment:
I'm wondering if we should not also modify `check_is_fitted` to be lenient with stateless estimator. Right now, one could expect to implement `__sklearn_is_fitted__` but I don't think this is part of the API per se. So I'm wondering if `check_is_fitted` should look at the tag `requires_fit`?

Reply:
Done via: https://github.com/scikit-learn/scikit-learn/pull/29880